### DISCUSSION TITLE
What does api.cache(true) exactly does in babel.config.js?

### DISCUSSION THREAD

**Discussion by shubhamdeodia** [babel/babel#12080] [Q&A] [answered]
Hi Team,

It would be really helpful, If you could help me understand the exact purpose of the api.cache(true) inside `babel.config.js`

What I have understood from the documentation is that to inhibit re-execution of the function which is exported from `babel.config.js` (**config function**), babel provides the config function api as an argument.

If the user, sets this `api.cache(true)`, then the result of this config function will be cached, and the config function will be executed only once. Thereby saving the compilation time by resuming the result from the cache.

Did I understood this correctly, or am I missing something?

Appreciate your help.

**Comment by nicolo-ribaudo** [comment, marked as answer]
`api.cache(true)` allows Babel to avoid re-evaluating your configuration factory function, giving a small speedup. Not only configs, but also plugins and presets can configure their own caching mechanisms to avoid being re-instantiated every time.
